### ISSUE TITLE
add option to tie transactions to user options as opposed to flex cookies

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#1934]
_Imported from trac ticket 1934.
Created by **mariam**
Opened in trac: Sep 13, 2012, 1:16:58 PM
Last modified in trac: Dec 23, 2015, 10:11:16 AM_
***
_by mariam:_
I don't know if Mariam has brought up this before, but we are having an issue with our thin-clients and PCs slowing down tremendously after extended periods of using LOFT.   We found that the way to make LOFT run quickly again is to clear the Cache on the clients.  If not done correctly, the cookies also get cleared.

This results in having the salespeople deciding what their store number and cashier number is when logging into LOFT for the first time. This also happens every time we get a new update.  As a result, we are having people entering the wrong store number and wrong cashier station numbers.  This will only get worse once we begin using credit cards with LOFT, since the cashier station will be MUCH MORE important in order to balance the money taken by the salespeople.

Bottom line:  Can we tie the salesperson to the store/cashier station?  This could be as an OPTION. Have each company decide whether to tie the physical station or the person to the store/cash drawer.

Thank you.

Jesús R. Capó, VP, CIO

**Comment by kentbower** [comment]
_**rneville** trac comment on Sep 6, 2013, 10:48:36 AM:_

This wont work. We have to have that information at the time they are authenticated and we haven't set or verified the user at that point in time, so it's too late in the process. The other tricky part about the way this works is that this idea would work fine if there was a persistent session being managed on the server but that isn't what's happening. I closed a similar ticket to this a few days ago for the same reason. Site and cash drawer are actually stored in memory in the Flex client.

As an alternative idea, what if we made it so that these values could be defaulted via URL parameters? That way retailers could define shortcuts in the OS that would point to something like http://eld.retailarchitects.com?site=01&drawer=00 which would default those selections on the login screen if they were not already set via a cookie. The user could still choose something different if they wanted to and needed to (and had the proper permissions to). Without some poking around, I'm not even sure I can make _that_ work.

This still makes the link between site and cash drawer a physical link to the PC in question and not to the person (unless you did something clever like roaming OS profiles such that the shortcut they had on their desktop was tied to their domain user/password or something like that).

Would this satisfy the requirement?

**Comment by kentbower** [comment]
_**rneville** trac status change on Sep 6, 2013, 10:48:36 AM:_ **waiting**
